app/controllers/deliveries_controller.rb

```diff
@@ -0,0 +1,63 @@
+class DeliveriesController < ApplicationController
+  def index
+    matching_deliveries = Delivery.all
+
+    @list_of_deliveries = matching_deliveries.order({ :created_at => :desc })
+
+    render({ :template => "deliveries/index" })
+  end
+
+  def show
+    the_id = params.fetch("path_id")
+
+    matching_deliveries = Delivery.where({ :id => the_id })
+
+    @the_delivery = matching_deliveries.at(0)
+
+    render({ :template => "deliveries/show" })
+  end
+
+  def create
+    the_delivery = Delivery.new
+    the_delivery.description = params.fetch("query_description")
+    the_delivery.supposed_to_arrive_on = params.fetch("query_supposed_to_arrive_on")
+    the_delivery.details = params.fetch("query_details")
+    the_delivery.user_id = params.fetch("query_user_id")
+    the_delivery.arrived = false
+
+    if the_delivery.valid?
+      the_delivery.save
+      redirect_to("/deliveries", { :notice => "Added to list." })
+    else
+      redirect_to("/deliveries", { :alert => the_delivery.errors.full_messages.to_sentence })
+    end
+  end
+
+  def update
+    the_id = params.fetch("path_id")
+    the_delivery = Delivery.where({ :id => the_id }).at(0)
+
+    the_delivery.description = params.fetch("query_description")
+    the_delivery.supposed_to_arrive_on = params.fetch("query_supposed_to_arrive_on")
+    the_delivery.details = params.fetch("query_details")
+    the_delivery.user_id = params.fetch("query_user_id")
```

Comment:
you dont necessarily need to update all of the columns when updating your delivery, you just need to make sure arrived switches to true :)